### DISCUSSION TITLE
`ProductVariant` barcode / GTIN support

### DISCUSSION THREAD

**Discussion by m-sanders** [Shopify/storefront-api-feedback#36] [Ideas & Requests]
Hello 

We've added [Product Schema.org structured data markup JSON-LD](https://support.google.com/merchants/answer/6386198?hl=en-GB) data to all of our product pages to keep Google happy. Exposing the globally unique product ID (like a GTIN, UPC, ISBN, etc) in the structured data schema.org markup is highly recommended by Google.

 From what I can tell, Shopify Storefront API `ProductVariant` type does not expose the `barcode` field (labelled in the admin as "Barcode (ISBN, UPC, GTIN, etc.)") and so we've had to add a `barcode` `PRODUCTVARIANT` metafield with the same data in order to expose the data to the client. This is problematic as we have the same data in two locations and our team forgets to sometimes add these values or update them. They also need to be the same value for the schema.org market to work correctly with Google Shopping, which cross references the value in the Shopping Feed (the native Barcode field value) with the scraped Schema.org data (the metafield value).

It would be ideal if the `ProductVariant` type could expose the `barcode` value the same as with themes:

```
{% for variant in product.variants %}
{{ variant.sku }}
{{ variant.barcode }}
{% endfor %}
```

**Comment by cocoahero** [comment]
That's a great suggestion @m-sanders. Out of curiosity, can you elaborate on what you are using the `barcode` value for in the context of a Storefront API client? Are you displaying them to customers?

**Reply by m-sanders** [reply]
We are using the `barcode` value to store the `GTIN`. We then use this value in the Schema.org structured data markup for bots like the Google Bot:

https://support.google.com/merchants/answer/6386198?hl=en-GB

The `barcode` value can be used in the theme but is not available to the Storefront API:

```
{% for variant in product.variants %}
{{ variant.sku }}
{{ variant.barcode }}
{% endfor %}
```

**Reply by m-sanders** [reply]
I have updated my original feature request to include the additional context.

**Reply by cocoahero** [reply]
Good news! The `ProductVariant.barcode` field is now available on the `unstable` version of the API. It is scheduled to be released to the next stable version which should be `2022-01`.

**Reply by m-sanders** [reply]
Fantastic news! Thanks @cocoahero.

**Reply by cocoahero** [reply]
Hey there @m-sanders - just wanted to follow up and confirm that the barcode field has been added to the `2022-01` release candidate. Thanks again!
